src/main/java/org/isf/sms/providers/gsm/GSMGatewayService.java

```diff
@@ -24,33 +24,29 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
-import java.util.Enumeration;
-import java.util.TooManyListenersException;
 
 import org.isf.sms.model.Sms;
 import org.isf.sms.providers.SmsSenderInterface;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.boot.web.server.PortInUseException;
 import org.springframework.stereotype.Component;
 
-import gnu.io.CommPortIdentifier;
-import gnu.io.PortInUseException;
-import gnu.io.SerialPort;
-import gnu.io.SerialPortEvent;
-import gnu.io.SerialPortEventListener;
+import com.fazecast.jSerialComm.SerialPort;
+import com.fazecast.jSerialComm.SerialPortDataListener;
+import com.fazecast.jSerialComm.SerialPortEvent;
 
 /**
  * @author Mwithi 03/feb/2014
  */
 @Component
-public class GSMGatewayService implements SmsSenderInterface, SerialPortEventListener {
+public class GSMGatewayService implements SmsSenderInterface, SerialPortDataListener {
 
 	public static final String SERVICE_NAME = "gsm-gateway-service";
 	private static final Logger LOGGER = LoggerFactory.getLogger(GSMGatewayService.class);
 	private static final String EOF = "\r";
 
-	private Enumeration< ? > portList;
-	private CommPortIdentifier portId;
+	private SerialPort[] portList;
```

Comment:
This variable is only used in the `initialize()` routine.  It should be defined as a local variable there and not a class level variable.